tests/baselines/reference/stringMappingOverPatternLiterals.errors.txt

```diff
@@ -0,0 +1,250 @@
+!!! error TS2322:   Type '`AA${string}`' is not assignable to type '"AA"'.
+        c = c_template;
+        ~
+!!! error TS2322: Type '`aa${Lowercase<string>}`' is not assignable to type '"aa"'.
+!!! error TS2322:   Type '`aa${string}`' is not assignable to type '"aa"'.
+    
+        // Additionally, all the template versions should be mutually incompatible (they describe differing sets)
+        a_template = b_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`AA${Uppercase<string>}`' is not assignable to type '`aA${string}`'.
+        a_template = c_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`aa${Lowercase<string>}`' is not assignable to type '`aA${string}`'.
+    
+        b_template = a_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`aA${string}`' is not assignable to type '`AA${Uppercase<string>}`'.
+        b_template = c_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`aa${Lowercase<string>}`' is not assignable to type '`AA${Uppercase<string>}`'.
+    
+        c_template = a_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`aA${string}`' is not assignable to type '`aa${Lowercase<string>}`'.
+        c_template = b_template;
+        ~~~~~~~~~~
+!!! error TS2322: Type '`AA${Uppercase<string>}`' is not assignable to type '`aa${Lowercase<string>}`'.
+    }
+    
+    // Raw string mapping assignability
+    
+    function f2(x1: string, x2: Uppercase<string>, x3: Lowercase<string>) {
+        // ok
+        x1 = x2;
+        x1 = x3;
+    
+        x2 = "ABC";
+        x3 = "abc";
+    
+        // should fail (sets do not match)
+        x2 = x1;
+        ~~
+!!! error TS2322: Type 'string' is not assignable to type 'Uppercase<string>'.
+        x2 = x3;
+        ~~
+!!! error TS2322: Type 'Lowercase<string>' is not assignable to type 'Uppercase<string>'.
+        x3 = x1;
+        ~~
+!!! error TS2322: Type 'string' is not assignable to type 'Lowercase<string>'.
+        x3 = x2;
+        ~~
+!!! error TS2322: Type 'Uppercase<string>' is not assignable to type 'Lowercase<string>'.
+    
+        x2 = "AbC";
+        ~~
+!!! error TS2322: Type 'string' is not assignable to type 'Uppercase<string>'.
+        x3 = "AbC";
+        ~~
+!!! error TS2322: Type 'string' is not assignable to type 'Lowercase<string>'.
+    }
+    
+    // Mappings over mappings
+    
+    function f3(
+        x1: Uppercase<string>,
+        x2: Uppercase<Uppercase<string>>,
+        x3: Uppercase<Lowercase<string>>) {
+    
+        // _ideally_ these would all be equivalent
+        x1 = x2;
+        x1 = x3;
+    
+        x2 = x1;
+        x2 = x3;
+    
+        // you'd think these were equivalent - the outer `Uppercase` conceptually
+        // makes the inner `Lowercase` effectively a noop - but that's not so;
+        // the german sharp s makes that not completely true (lowercases to ss,
+        // which then uppercases to SS), so arbitrary nestings of mappings make differing sets!

```

Comment:
Ugh.

But is it true that `Uppercase<Lowercase<string>>` is just `Uppercase<string>`?

Does last one wins apply, or is there something 'bout Turkish ı?

Reply:
> But is it true that `Uppercase<Lowercase<string>>` is just `Uppercase<string>`?

Nope.

> Does last one wins apply,

Nope, differing mappings stack forever. Only, eg, `Uppercase<Uppercase<...>>`  will collapse the multiple consecutive `Uppercase`s.

> is there something 'bout Turkish ı?

Yep.

`"ı".toLowerCase().toUpperCase() !== "ı"`

Reply:
I realized it after the fact, I just wanted to comment with a rhyme.